Index: elua/src/platform/avr32/MIZAR32/mizar32_conf.h
===================================================================
--- elua.orig/src/platform/avr32/MIZAR32/mizar32_conf.h	2011-03-22 04:50:12.268018002 +0100
+++ elua/src/platform/avr32/MIZAR32/mizar32_conf.h	2011-03-22 05:09:26.224018001 +0100
@@ -17,7 +17,7 @@
 //#define BUILD_XMODEM
 //#define BUILD_SHELL
 //#define BUILD_ROMFS
-//#define BUILD_TERM
+#define BUILD_TERM
 #define BUILD_CON_GENERIC
 //#define BUILD_RPC
 #define BUF_ENABLE_UART
@@ -66,23 +66,18 @@
 #endif
 
 #define LUA_PLATFORM_LIBS_ROM\
-  _ROM( AUXLIB_PD, luaopen_pd, pd_map )\
-  _ROM( AUXLIB_UART, luaopen_uart, uart_map )\
-  _ROM( AUXLIB_PIO, luaopen_pio, pio_map )\
-  _ROM( AUXLIB_TMR, luaopen_tmr, tmr_map )\
-  _ROM( AUXLIB_TERM, luaopen_term, term_map )\
+  _ROM( AUXLIB_BIT, luaopen_bit, bit_map )\
   _ROM( AUXLIB_CPU, luaopen_cpu, cpu_map )\
   _ROM( AUXLIB_ELUA, luaopen_elua, elua_map )\
-  RPCLINE\
-  _ROM( AUXLIB_BIT, luaopen_bit, bit_map )\
   _ROM( AUXLIB_PACK, luaopen_pack, pack_map )\
-  _ROM( LUA_MATHLIBNAME, luaopen_math, math_map )
-
-/* Minimal ROM modules, to fit in 120KB */
-#undef  LUA_PLATFORM_LIBS_ROM
-#define LUA_PLATFORM_LIBS_ROM\
+  _ROM( AUXLIB_PD, luaopen_pd, pd_map )\
   _ROM( AUXLIB_PIO, luaopen_pio, pio_map )\
+  _ROM( AUXLIB_SPI, luaopen_spi, spi_map )\
+  _ROM( AUXLIB_TERM, luaopen_term, term_map )\
   _ROM( AUXLIB_TMR, luaopen_tmr, tmr_map )\
+  _ROM( AUXLIB_UART, luaopen_uart, uart_map )\
+  RPCLINE\
+
 
 // *****************************************************************************
 // Configuration data
