Index: elua/src/newlib/genstd.c
===================================================================
--- elua.orig/src/newlib/genstd.c	2011-04-23 19:40:33.335233003 +0200
+++ elua/src/newlib/genstd.c	2011-04-23 19:41:24.503233002 +0200
@@ -44,8 +44,11 @@
     c = std_get_char_func( STD_INFINITE_TIMEOUT );
     switch( c )
     {
+// If there's no shell, quitting Lua makes no sense.
+#ifdef BUILD_SHELL
     case STD_CTRLZ_CODE:  // End of file
       return 0;
+#endif
 
     case -1:              // UART overrun/failure - return chars so far
       return i;
