Index: elua/src/buf.c
===================================================================
--- elua.orig/src/buf.c	2011-04-23 19:16:54.795233002 +0200
+++ elua/src/buf.c	2011-04-23 19:17:39.487233002 +0200
@@ -123,7 +123,6 @@
     return PLATFORM_ERR;    
   if( pbuf->count > BUF_REALSIZE( pbuf ) )
   {
-    fprintf( stderr, "[ERROR] Buffer overflow on resid=%d, resnum=%d!\n", resid, resnum );
     return PLATFORM_ERR; 
   }
   DUFF_DEVICE_8( BUF_REALDSIZE( pbuf ),  *d++ = *s++ );
