Index: elua-svn/src/elua_mmc.c
===================================================================
--- elua-svn.orig/src/elua_mmc.c	2010-12-17 16:55:54.575835000 +0100
+++ elua-svn/src/elua_mmc.c	2010-12-17 16:59:04.111835000 +0100
@@ -35,16 +35,16 @@
 static
 void SELECT (void)
 {
-  platform_pio_op( MMCFS_CS_PORT , ( ( u32 ) 1 << MMCFS_CS_PIN ), PLATFORM_IO_PIN_CLEAR ); 
-	//platform_spi_select( MMCFS_SPI_NUM, PLATFORM_SPI_SELECT_ON ); // select sd card pin
+  //platform_pio_op( MMCFS_CS_PORT , ( ( u32 ) 1 << MMCFS_CS_PIN ), PLATFORM_IO_PIN_CLEAR );
+  platform_spi_select( MMCFS_SPI_NUM, PLATFORM_SPI_SELECT_ON ); // select sd card pin
 }
 
 // de-asserts the CS pin to the card
 static
 void DESELECT (void)
 {
-  platform_pio_op( MMCFS_CS_PORT, ( ( u32 ) 1 << MMCFS_CS_PIN ), PLATFORM_IO_PIN_SET );
-	//platform_spi_select( MMCFS_SPI_NUM, PLATFORM_SPI_SELECT_OFF ); // deselect sd card pin
+  //platform_pio_op( MMCFS_CS_PORT, ( ( u32 ) 1 << MMCFS_CS_PIN ), PLATFORM_IO_PIN_SET );
+  platform_spi_select( MMCFS_SPI_NUM, PLATFORM_SPI_SELECT_OFF ); // deselect sd card pin
 }
 
 
