Index: elua-svn/src/platform/avr32/MIZAR32/mizar32_conf.h
===================================================================
--- elua-svn.orig/src/platform/avr32/MIZAR32/mizar32_conf.h	2011-01-15 23:42:38.226154000 +0100
+++ elua-svn/src/platform/avr32/MIZAR32/mizar32_conf.h	2011-01-15 23:43:56.326154000 +0100
@@ -63,23 +63,15 @@
 #endif
 
 #define LUA_PLATFORM_LIBS_ROM\
-  _ROM( AUXLIB_PD, luaopen_pd, pd_map )\
-  _ROM( AUXLIB_UART, luaopen_uart, uart_map )\
-  _ROM( AUXLIB_PIO, luaopen_pio, pio_map )\
-  _ROM( AUXLIB_TMR, luaopen_tmr, tmr_map )\
-  _ROM( AUXLIB_TERM, luaopen_term, term_map )\
+  _ROM( AUXLIB_BIT, luaopen_bit, bit_map )\
   _ROM( AUXLIB_CPU, luaopen_cpu, cpu_map )\
   _ROM( AUXLIB_ELUA, luaopen_elua, elua_map )\
-  RPCLINE\
-  _ROM( AUXLIB_BIT, luaopen_bit, bit_map )\
-  _ROM( AUXLIB_PACK, luaopen_pack, pack_map )\
-  _ROM( LUA_MATHLIBNAME, luaopen_math, math_map )
-
-/* Minimal ROM modules, to fit in 120KB */
-#undef  LUA_PLATFORM_LIBS_ROM
-#define LUA_PLATFORM_LIBS_ROM\
+  _ROM( AUXLIB_PD, luaopen_pd, pd_map )\
   _ROM( AUXLIB_PIO, luaopen_pio, pio_map )\
+  _ROM( AUXLIB_SPI, luaopen_spi, spi_map )\
   _ROM( AUXLIB_TMR, luaopen_tmr, tmr_map )\
+  _ROM( AUXLIB_UART, luaopen_uart, uart_map )\
+  RPCLINE\
 
 // *****************************************************************************
 // Configuration data
